src/plugins/data/common/query/filter_manager/compare_filters.ts

```diff
@@ -44,6 +46,7 @@ const mapFilter = (
 ) => {
   const cleaned: FilterMeta = omit(filter, excludedAttributes);
 
+  if (comparators.index) cleaned.index = filter.meta?.index;
```

Comment:
I think we have a pretty good unit test coverage for this function. Should we for sanity consider also covering this new param?